experimental/yarpl/src/yarpl/flowable/utils/SubscriptionHelper.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include "SubscriptionHelper.h"
+#include <atomic>
+#include <climits>
+
+namespace yarpl {
+namespace flowable {
+namespace internal {
+int64_t SubscriptionHelper::addCredits(
+    std::atomic<std::int64_t>* current,
+    int64_t n) {
+  for (;;) {
+    int64_t r = current->load();
+    if (r == INT64_MAX) {
+      return INT64_MAX;
+    }
+    if (n <= 0) {
+      // do nothing, return existing unmodified value
+      return r;
+    }
+    long u = r + n;
```

Comment:
long type will be 32 bit number on 32 bit architecture. You probably want to be more specific and use int64_t or "long long"